### PR TITLE
Added support for authenticated encryption

### PR DESCRIPTION
**Pull request by awoie**
Added ECDH-1PU+XC20PKW, AuthEncrypter/Decrypter and AnonEncrypter/Decrypter

### CODE REVIEW

**Comment by awoie** [comment]
@oed pls review again

**Review by mirceanis — APPROVED** [review]
looks good to me

File: src/xc20pEncryption.ts [hunk 1]
```diff
@@ -9,6 +9,69 @@ import type { VerificationMethod, Resolvable } from 'did-resolver'
...
+}
+
+/** 
+ * Recommended encrypter for anonymous encryption (i.e. no sender authentication).
+ * Uses ECDH-ES+XC20PKW [v2](https://tools.ietf.org/html/draft-amringer-jose-chacha-02).
+ * 
+ * NOTE: ECDH-ES+XC20PKW is a proposed draft in IETF and not a standard yet and
+ * is subject to change as new revisions or until the offical CFRG specification is released. 
+ */
+export function AnonEncrypter(publicKey: Uint8Array,
```
Comment (mirceanis) [inline src/xc20pEncryption.ts:46]:
```suggestion
export function createAnonEncrypter(publicKey: Uint8Array, 
```

File: src/xc20pEncryption.ts [hunk 2]
```diff
@@ -9,6 +9,69 @@ import type { VerificationMethod, Resolvable } from 'did-resolver'
...
+/** 
+ * Recommended decrypter for authenticated encryption (i.e. sender authentication and requires
+ * sender public key to decrypt the data).
+ * Uses ECDH-1PU [v3](https://tools.ietf.org/html/draft-madden-jose-ecdh-1pu-03) and 
+ * XC20PKW [v2](https://tools.ietf.org/html/draft-amringer-jose-chacha-02).
+ * 
+ * NOTE: ECDH-1PU and XC20PKW are proposed drafts in IETF and not a standard yet and
+ * are subject to change as new revisions or until the offical CFRG specification are released. 
+ */
+export function AuthDecrypter(recipientSecretKey: Uint8Array, senderPublicKey: Uint8Array): Decrypter {
```
Comment (mirceanis) [inline src/xc20pEncryption.ts:60]:
```suggestion
export function createAuthDecrypter(recipientSecretKey: Uint8Array, senderPublicKey: Uint8Array): Decrypter {
```

File: src/xc20pEncryption.ts [hunk 3]
```diff
@@ -9,6 +9,69 @@ import type { VerificationMethod, Resolvable } from 'did-resolver'
...
+}
+
+/** 
+ * Recommended decrypter for anonymous encryption (i.e. no sender authentication).
+ * Uses ECDH-ES+XC20PKW [v2](https://tools.ietf.org/html/draft-amringer-jose-chacha-02).
+ * 
+ * NOTE: ECDH-ES+XC20PKW is a proposed draft in IETF and not a standard yet and
+ * is subject to change as new revisions or until the offical CFRG specification is released. 
+ */
+export function AnonDecrypter(secretKey: Uint8Array): Decrypter {
```
Comment (mirceanis) [inline src/xc20pEncryption.ts:71]:
```suggestion
export function createAnonDecrypter(secretKey: Uint8Array): Decrypter {
```

**Review by mirceanis — CHANGES_REQUESTED** [review]
I added some suggestions for what could be slightly better names for some methods and a reminder to also export the new methods in `index.ts`

**Review by mirceanis — APPROVED** [review]
looks good to me

**Review by oed — APPROVED** [review]
lgtm. Would be nice to have Some API documentation for the JWE stuff though!

**Pull request merged**